codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -421,7 +421,11 @@ protected void writeMiddlewarePayloadSerializerDelegator(
 
                     } else if (payloadShape.isStringShape()) {
                         writer.addUseImports(SmithyGoDependency.STRINGS);
-                        writer.write("payload := strings.NewReader(*$L)", s);
+                        if (payloadShape.hasTrait(EnumTrait.class)) {
+                            writer.write("payload := strings.NewReader(string($L))", s);
+                        } else {
+                            writer.write("payload := strings.NewReader(*$L)", s);
```

Comment:
Good find. I think this logic must have been duplicated elsewhere. 

Also something phjordon@ suggested to me few weeks back - that I liked.

```
String v = String.format("payload := strings.NewReader(%s)", payloadShape.hasTrait(EnumTrait.class)? "string($L)": "*$L");
writer.write(v);
```